src/fsharp/Optimizer.fs

```diff
@@ -1664,12 +1664,10 @@ let TryDetectQueryQuoteAndRun cenv (expr:Expr) =
         None
                 
 let IsSystemStringConcat (methRef: ILMethodRef) =
-    let isConcat =
-        methRef.Name = "Concat" && methRef.DeclaringTypeRef.FullName = "System.String" && 
-        (methRef.ArgTypes |> List.forall(fun ilty -> ilty.BasicQualifiedName = "System.String") || 
-         (methRef.ArgTypes.Length = 1 && methRef.ArgTypes.Head.BasicQualifiedName = "System.String[]")) &&
-        methRef.ReturnType.BasicQualifiedName = "System.String"
-    isConcat
+    methRef.Name = "Concat" && methRef.DeclaringTypeRef.FullName = "System.String" && 
```

Comment:
Just wondering: is this allocating "System.String" 3 times?

Reply:
No, a literal "System.String" is only allocated once and re-used.

Reply:
More info: https://blog.maartenballiauw.be/post/2016/11/15/exploring-memory-allocation-and-strings.html Nice blog post.